server/player/handler.go

```diff
@@ -55,6 +55,9 @@ type Handler interface {
 	// HandleSkinChange handles the player changing their skin. ctx.Cancel() may be called to cancel the skin
 	// change.
 	HandleSkinChange(ctx *event.Context, skin *skin.Skin)
+	// HandleFireExtinguish handles the player extinguishing a fire at a specific position. ctx.Cancel() may
+	// be called to cancel the fire being extinguished.
+	HandleFireExtinguish(ctx *event.Context, pos cube.Pos)
 	// HandleStartBreak handles the player starting to break a block at the position passed. ctx.Cancel() may
 	// be called to stop the player from breaking the block completely.
 	HandleStartBreak(ctx *event.Context, pos cube.Pos)
```

Comment:
is there a reason for pushing cube.Pos? this seems un-needed

Reply:
I see it as necessary when wanting to cancel the fire extinguishment only in some positions, you cannot use the player's position, since both positions are different than expected.